reference-implementation/to-upstream-wpts/writable-streams/close.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+promise_test(t => {
+  const ws = new WritableStream({
+    close() {
+      return 'Hello';
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  const closePromise = writer.close('a');
+  return closePromise.then(value => assert_equals(value, undefined, 'fulfillment value must be undefined'));
+}, 'fulfillment value of ws.close() call must be undefined even if the underlying sink returns a non-undefined ' +
+    'value');
+
+async_test(t => {
+  const thrownError = new Error('throw me');
+  const ws = new WritableStream({
+    close() {
+      throw thrownError;
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  promise_rejects(
+      t, thrownError, writer.close(), 'close promise', 'close promise should be rejected with the thrown error');
```

Comment:
I think it's ok. I added promise_rejects in this file as preparation for migration to testharness.js.